src/OrleansRuntime/Placement/StatelessWorkerDirector.cs

```diff
@@ -12,6 +12,26 @@ internal class StatelessWorkerDirector : IPlacementDirector<StatelessWorkerPlace
         public Task<PlacementResult> OnSelectActivation(
             PlacementStrategy strategy, GrainId target, IPlacementRuntime context)
         {
+            return Task.FromResult(SelectActivationCore(strategy, target, context));
+        }
+
+        public bool TrySelectActivationSynchronously(
+            PlacementStrategy strategy, GrainId target, IPlacementRuntime context, out PlacementResult placementResult)
+        {
+            placementResult = SelectActivationCore(strategy, target, context);
+            // stateless workers are always local, and  request for creation of a new activation will be created by PlacementDirectorsManager 
+            // in case of null returned by SelectActivationCore  
+            return true;
```

Comment:
OK about Try with bool. Lets not do any more changes about OnSelectActivation now, in this CL.